users/test_update_authorization.py

```diff
@@ -0,0 +1,112 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import mongomock
+import app
+
+AUTHORIZED_USER = "authorized-user"
+NON_AUTHORIZED_USER = "non-authorized-user"
+MALFORMATTED_IN_DB_USER = "this-user-has-no-'is_organizer'_db_field"
+MISSING_USER = "not-a-user-in-the-database"
+
+
+class TestUpdateAuthorization(unittest.TestCase):
+    """Test /authorization endpoint of users service."""
+
+    def setUp(self):
+        """Seed mock DB for testing"""
+        self.mock_collection = mongomock.MongoClient().db.collection
+        mock_data = [
+            {"user_id": AUTHORIZED_USER,
+             "name": AUTHORIZED_USER,
+             "is_organizer": True},
+            {"user_id": NON_AUTHORIZED_USER,
+             "name": NON_AUTHORIZED_USER,
+             "is_organizer": False},
+            {"user_id": MALFORMATTED_IN_DB_USER}
+        ]
+        self.mock_collection.insert_many(mock_data)
+
+    def test_change_authorization(self):
+        """Can change the authorization to the opposite value."""
+        # True to False
+        self.assertTrue(self.mock_collection.find_one(
+            {"user_id": AUTHORIZED_USER})["is_organizer"])
+        app.update_user_authorization_in_db(
+            AUTHORIZED_USER, False, self.mock_collection)
+        self.assertFalse(self.mock_collection.find_one(
+            {"user_id": AUTHORIZED_USER})["is_organizer"])
+        # False to True
+        self.assertFalse(self.mock_collection.find_one(
+            {"user_id": NON_AUTHORIZED_USER})["is_organizer"])
+        app.update_user_authorization_in_db(
+            NON_AUTHORIZED_USER, True, self.mock_collection)
+        self.assertTrue(self.mock_collection.find_one(
+            {"user_id": NON_AUTHORIZED_USER})["is_organizer"])
+
+    def test_keep_authorization_the_same(self):
+        """Can trivially change the authorization to the same value."""
+        # True to True
+        self.assertTrue(self.mock_collection.find_one(
+            {"user_id": AUTHORIZED_USER})["is_organizer"])
+        app.update_user_authorization_in_db(
+            AUTHORIZED_USER, True, self.mock_collection)
+        self.assertTrue(self.mock_collection.find_one(
+            {"user_id": AUTHORIZED_USER})["is_organizer"])
+        # False to False
+        self.assertFalse(self.mock_collection.find_one(
+            {"user_id": NON_AUTHORIZED_USER})["is_organizer"])
+        app.update_user_authorization_in_db(
+            NON_AUTHORIZED_USER, False, self.mock_collection)
+        self.assertFalse(self.mock_collection.find_one(
+            {"user_id": NON_AUTHORIZED_USER})["is_organizer"])
+
+    def test_set_authorization_of_malformed_user(self):
```

Comment:
Do you expect to have malformed users in the DB?

Reply:
Not really, but I wanted to make sure the behaviour is defined safely in that case. @evankanderson ?

Reply:
Cool, thanks for the clarification.